server/apps/publish/formatters/ninjs_formatter.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+import json
+
+from apps.publish.formatters import Formatter
+import superdesk
+from superdesk.errors import FormatterError
+from bson import json_util
+
+
+class NINJSFormatter(Formatter):
+    """
+    NINJS Formatter
+    """
+    direct_copy_properties = ['versioncreated', 'usageterms', 'subject', 'language', 'headline',
+                              'urgency', 'pubstatus', 'mimetype', 'renditions', 'place', 'located']
+
+    def format(self, article, destination, selector_codes=None):
+        try:
+            pub_seq_num = superdesk.get_resource_service('output_channels').generate_sequence_number(destination)
+
+            ninjs = {}
+            ninjs['_id'] = article['_id']
+            ninjs['version'] = str(article['version'])
+            ninjs['type'] = self._get_type(article)
+            try:
+                ninjs['byline'] = self._get_byline(article)
+            except:
+                pass
+            for copy_property in self.direct_copy_properties:
+                if copy_property in article:
+                    ninjs[copy_property] = article[copy_property]
+            if article['type'] == 'composite':
+                article['associations'] = self._get_associations(article)
+
+            return pub_seq_num, json.dumps(ninjs, default=json_util.default)
```

Comment:
There should also be _created and _updated fields here, Eve uses them to handle conditional HTTP requests (the `If-Modified-Since` header).

There is a [separate ticket](https://dev.sourcefabric.org/browse/SD-2429) for that, but we might as well resolve it here while we are adding this formatting functionality.